src/test/mir-opt/validate_3.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// ignore-tidy-linelength
+// compile-flags: -Z verbose -Z mir-emit-validate
+
+struct Test {
+    x: i32
+}
+
+fn foo(_x: &i32) {}
+
+fn main() {
+    let t = Test { x: 0 };
+    let t = &t;
+    foo(&t.x);
```

Comment:
These tests seem good, if basic. I wonder if we want to test any of the more subtle scenarios...? Maybe this is good enough for now. These kinds of tests *are* somewhat brittle -- the main thing I want is some way to ensure we are not bitrotting your code when we rebase. To that end, maybe something that tests a case where we have to "reacquire" the result from a call as well would be a good idea?

e.g., `let x = vec.split_at_mut(3);`